### ISSUE TITLE
Wrong validation where one char has wrong casing

### ISSUE THREAD

**Issue by pnowosie** [PumpkinSeed/eip55-rs#1]
Please consider the following snippet [I've tried it on replit](https://replit.com/@pnowo/eip55-validator#src/main.rs):

```rust
// correct checksum
const ADDRESS: &str = "0x000000000000000000000000000000000000dEaD";

fn main() {
  // NOTE: This is incorrect, valid form is `..dEaD` as above
  assert_eq!(
    checksum("0x000000000000000000000000000000000000dEAD"),
    "0x000000000000000000000000000000000000dEAD");

  // and hence wrong result here
  assert_eq!(
    validate_address("0x000000000000000000000000000000000000dEAD"),
    true);
}
```

**Comment by PumpkinSeed** [comment]
Sorry for the very late answer, I had a tons of notifications and I missed that one. I sent a PR to resolve that issue. Thanks for the report.

**Comment by PumpkinSeed** [comment]
I released v0.2.0 to crates.io which has this update.
